<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>RNA-seq data analysis for H3mm7</title>

<script src="RNA-seq_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="RNA-seq_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="RNA-seq_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="RNA-seq_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="RNA-seq_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="RNA-seq_files/navigation-1.1/tabsets.js"></script>
<link href="RNA-seq_files/highlightjs-1.1/default.css" rel="stylesheet" />
<script src="RNA-seq_files/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="fluid-row" id="header">



<h1 class="title toc-ignore">RNA-seq data analysis for H3mm7</h1>

</div>


<div id="prerequisites" class="section level2">
<h2>Prerequisites</h2>
<pre class="r"><code>library(clusterProfiler)
library(org.Mm.eg.db)
library(DT)
library(knitr)
library(data.table)
library(dtplyr)
library(dplyr)
source(&quot;RScript/DESeq2Workflow.R&quot;)</code></pre>
<pre class="r"><code>print(sessionInfo(),locale=FALSE)</code></pre>
<pre><code>## R version 3.3.2 (2016-10-31)
## Platform: x86_64-pc-linux-gnu (64-bit)
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] DESeq2_1.14.1              SummarizedExperiment_1.4.0
##  [3] GenomicRanges_1.26.4       GenomeInfoDb_1.10.3       
##  [5] ggplot2_2.2.1              RColorBrewer_1.1-2        
##  [7] dplyr_0.7.3                dtplyr_0.0.2              
##  [9] data.table_1.10.4          knitr_1.17                
## [11] DT_0.2                     org.Mm.eg.db_3.4.0        
## [13] AnnotationDbi_1.36.2       IRanges_2.8.2             
## [15] S4Vectors_0.12.2           Biobase_2.34.0            
## [17] BiocGenerics_0.20.0        clusterProfiler_3.2.14    
## [19] DOSE_3.0.10               
## 
## loaded via a namespace (and not attached):
##  [1] tidyr_0.7.1         bit64_0.9-7         splines_3.3.2      
##  [4] Formula_1.2-2       assertthat_0.2.0    DO.db_2.9          
##  [7] latticeExtra_0.6-28 blob_1.1.0          yaml_2.1.14        
## [10] RSQLite_2.0         backports_1.1.0     lattice_0.20-35    
## [13] glue_1.1.1          digest_0.6.12       checkmate_1.8.3    
## [16] XVector_0.14.1      qvalue_2.6.0        colorspace_1.3-2   
## [19] htmltools_0.3.6     Matrix_1.2-11       plyr_1.8.4         
## [22] XML_3.98-1.9        pkgconfig_2.0.1     genefilter_1.56.0  
## [25] zlibbioc_1.20.0     xtable_1.8-2        purrr_0.2.3        
## [28] GO.db_3.4.0         scales_0.5.0        BiocParallel_1.8.2 
## [31] annotate_1.52.1     htmlTable_1.9       tibble_1.3.4       
## [34] nnet_7.3-12         lazyeval_0.2.0      survival_2.41-3    
## [37] magrittr_1.5        memoise_1.1.0       evaluate_0.10.1    
## [40] foreign_0.8-69      tools_3.3.2         stringr_1.2.0      
## [43] locfit_1.5-9.1      munsell_0.4.3       cluster_2.0.6      
## [46] bindrcpp_0.2        rlang_0.1.2         grid_3.3.2         
## [49] RCurl_1.95-4.8      htmlwidgets_0.9     igraph_1.1.2       
## [52] bitops_1.0-6        base64enc_0.1-3     rmarkdown_1.6      
## [55] gtable_0.2.0        DBI_0.7             reshape2_1.4.2     
## [58] R6_2.2.2            gridExtra_2.3       bit_1.1-12         
## [61] bindr_0.1           fastmatch_1.1-0     Hmisc_4.0-3        
## [64] fgsea_1.0.2         rprojroot_1.2       GOSemSim_2.0.4     
## [67] stringi_1.1.5       Rcpp_0.12.12        geneplotter_1.52.0 
## [70] rpart_4.1-11        acepack_1.4.1</code></pre>
<div id="helper-functions" class="section level3">
<h3>Helper functions</h3>
<pre class="r"><code>modTbl &lt;- function(x,...) x %&gt;%
  mutate(log10padj=log10(padj)) %&gt;%
  dplyr::select(cluster,1:3,log2FC=log2FoldChange,log10padj) %&gt;%
  inner_join(e2g,.,by=&quot;ens_gene&quot;) %&gt;% arrange(log10padj)

showDT &lt;- function(x,...) x %&gt;% modTbl %&gt;% 
  datatable(filter=&quot;top&quot;,rownames=FALSE,...) %&gt;%
  formatRound(4:9,3)

tlab &lt;- function(res,alpha=0.1,thresh=0)
  c(&quot;LT&quot;,&quot;NS&quot;,&quot;GT&quot;)[with(res,
    (padj &lt; alpha &amp; baseMean &gt; thresh)*sign(log2FoldChange))+2
  ]

degLabs &lt;- function(dds,contrast,alpha=0.1,thresh=100)
  dds %&gt;% results(contrast=contrast) %&gt;% tlab(alpha,thresh)

tablemd &lt;- function(x,...) knitr::kable(x,format=&quot;markdown&quot;,...)</code></pre>
</div>
<div id="load-biomart-annotation" class="section level3">
<h3>Load bioMart annotation</h3>
<pre class="r"><code>library(biomaRt)
ensembl &lt;- useMart(&quot;ENSEMBL_MART_ENSEMBL&quot;,host=&quot;asia.ensembl.org&quot;)
mart &lt;- useDataset(&quot;mmusculus_gene_ensembl&quot;,mart=ensembl)
e2g &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;,&quot;external_gene_name&quot;,&quot;gene_biotype&quot;), mart = mart)
e2g &lt;- dplyr::rename(e2g, ens_gene = ensembl_gene_id, ext_gene = external_gene_name, biotype = gene_biotype)</code></pre>
</div>
</div>
<div id="h3mm7ko-in-c2c12" class="section level2">
<h2>H3mm7KO in C2C12</h2>
<pre class="r"><code>defko &lt;- getdef(&quot;data/table_C2C12RNAseq.txt&quot;)
ddsko &lt;- getdds(&quot;data/countGENCODEvM9_C2C12RNAseq.edit.txt&quot;,defko,~time+cond)</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre class="r"><code>rldko &lt;- rlog(ddsko,blind=FALSE)
tablemd(colData(ddsko))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">cell</th>
<th align="left">clone</th>
<th align="left">time</th>
<th align="left">cond</th>
<th align="left">rep</th>
<th align="right">sizeFactor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C2C12.WT1.G</td>
<td align="left">C2C12</td>
<td align="left">WT1</td>
<td align="left">G</td>
<td align="left">WT</td>
<td align="left">Rep1</td>
<td align="right">3.3203216</td>
</tr>
<tr class="even">
<td align="left">C2C12.WT2.G</td>
<td align="left">C2C12</td>
<td align="left">WT2</td>
<td align="left">G</td>
<td align="left">WT</td>
<td align="left">Rep2</td>
<td align="right">0.5707238</td>
</tr>
<tr class="odd">
<td align="left">C2C12.3A3.G</td>
<td align="left">C2C12</td>
<td align="left">3A3</td>
<td align="left">G</td>
<td align="left">KO</td>
<td align="left">Rep1</td>
<td align="right">0.8365059</td>
</tr>
<tr class="even">
<td align="left">C2C12.3E4.G</td>
<td align="left">C2C12</td>
<td align="left">3E4</td>
<td align="left">G</td>
<td align="left">KO</td>
<td align="left">Rep2</td>
<td align="right">0.8409781</td>
</tr>
<tr class="odd">
<td align="left">C2C12.WT1.D24</td>
<td align="left">C2C12</td>
<td align="left">WT1</td>
<td align="left">D24</td>
<td align="left">WT</td>
<td align="left">Rep1</td>
<td align="right">3.0602057</td>
</tr>
<tr class="even">
<td align="left">C2C12.WT2.D24</td>
<td align="left">C2C12</td>
<td align="left">WT2</td>
<td align="left">D24</td>
<td align="left">WT</td>
<td align="left">Rep2</td>
<td align="right">0.3199982</td>
</tr>
<tr class="odd">
<td align="left">C2C12.3A3.D24</td>
<td align="left">C2C12</td>
<td align="left">3A3</td>
<td align="left">D24</td>
<td align="left">KO</td>
<td align="left">Rep1</td>
<td align="right">0.8484708</td>
</tr>
<tr class="even">
<td align="left">C2C12.3E4.D24</td>
<td align="left">C2C12</td>
<td align="left">3E4</td>
<td align="left">D24</td>
<td align="left">KO</td>
<td align="left">Rep2</td>
<td align="right">0.6981897</td>
</tr>
<tr class="odd">
<td align="left">C2C12.WT1.D48</td>
<td align="left">C2C12</td>
<td align="left">WT1</td>
<td align="left">D48</td>
<td align="left">WT</td>
<td align="left">Rep1</td>
<td align="right">2.8720869</td>
</tr>
<tr class="even">
<td align="left">C2C12.WT2.D48</td>
<td align="left">C2C12</td>
<td align="left">WT2</td>
<td align="left">D48</td>
<td align="left">WT</td>
<td align="left">Rep2</td>
<td align="right">0.6347962</td>
</tr>
<tr class="odd">
<td align="left">C2C12.3A3.D48</td>
<td align="left">C2C12</td>
<td align="left">3A3</td>
<td align="left">D48</td>
<td align="left">KO</td>
<td align="left">Rep1</td>
<td align="right">0.7034159</td>
</tr>
<tr class="even">
<td align="left">C2C12.3E4.D48</td>
<td align="left">C2C12</td>
<td align="left">3E4</td>
<td align="left">D48</td>
<td align="left">KO</td>
<td align="left">Rep2</td>
<td align="right">0.8979018</td>
</tr>
<tr class="odd">
<td align="left">C2C12.WT1.D72</td>
<td align="left">C2C12</td>
<td align="left">WT1</td>
<td align="left">D72</td>
<td align="left">WT</td>
<td align="left">Rep1</td>
<td align="right">2.3229792</td>
</tr>
<tr class="even">
<td align="left">C2C12.WT2.D72</td>
<td align="left">C2C12</td>
<td align="left">WT2</td>
<td align="left">D72</td>
<td align="left">WT</td>
<td align="left">Rep2</td>
<td align="right">0.8415663</td>
</tr>
<tr class="odd">
<td align="left">C2C12.3A3.D72</td>
<td align="left">C2C12</td>
<td align="left">3A3</td>
<td align="left">D72</td>
<td align="left">KO</td>
<td align="left">Rep1</td>
<td align="right">0.8621735</td>
</tr>
<tr class="even">
<td align="left">C2C12.3E4.D72</td>
<td align="left">C2C12</td>
<td align="left">3E4</td>
<td align="left">D72</td>
<td align="left">KO</td>
<td align="left">Rep2</td>
<td align="right">0.8022867</td>
</tr>
</tbody>
</table>
<pre class="r"><code>set.seed(333)
resko &lt;- results(ddsko,contrast=c(&quot;cond&quot;,&quot;KO&quot;,&quot;WT&quot;))
tblko &lt;- timecoursePlots(resko,rldko,k=6,showsd=TRUE)</code></pre>
<pre><code>## Warning in lda.default(x, grouping, ...): variables are collinear</code></pre>
<p><img src="RNA-seq_files/figure-html/results_H3mm7KO-1.png" /><!-- --><img src="RNA-seq_files/figure-html/results_H3mm7KO-2.png" /><!-- --><img src="RNA-seq_files/figure-html/results_H3mm7KO-3.png" /><!-- --></p>
<pre class="r"><code>p &lt;- ggplot(tblko,aes(cluster,baseMean,fill=cluster)) + 
  geom_violin() + scale_y_log10() + theme_bw() +
  scale_fill_brewer(palette=&quot;Set2&quot;)
print(p)</code></pre>
<p><img src="RNA-seq_files/figure-html/results_H3mm7KO-4.png" /><!-- --></p>
<pre class="r"><code>B &lt;- with(tblko %&gt;% modTbl, table(biotype,cluster))
op &lt;- par(mai=c(.5,2.2,.5,.5))
barplot(t(B[order(-rowSums(B))[1:6],]),beside=TRUE,col=brewer.pal(6,&quot;Set2&quot;),legend.text = colnames(B),las=1,horiz = TRUE)</code></pre>
<p><img src="RNA-seq_files/figure-html/results_H3mm7KO-5.png" /><!-- --></p>
<pre class="r"><code>par(op)
#showDT(tblko)</code></pre>
<pre class="r"><code>set.seed(333)
tblks &lt;- timecoursePlots(resko[resko$baseMean&gt;100,],rldko,k=6)</code></pre>
<pre><code>## Warning in lda.default(x, grouping, ...): variables are collinear</code></pre>
<p><img src="RNA-seq_files/figure-html/resultsH3mm7Sel-1.png" /><!-- --><img src="RNA-seq_files/figure-html/resultsH3mm7Sel-2.png" /><!-- --><img src="RNA-seq_files/figure-html/resultsH3mm7Sel-3.png" /><!-- --></p>
<pre class="r"><code>#showDT(tblks)</code></pre>
<div id="gsea" class="section level4">
<h4>GSEA</h4>
<pre class="r"><code>gsks &lt;- bitr(tblks$ens_gene,fromType = &quot;ENSEMBL&quot;,toType = &quot;ENTREZID&quot;,org.Mm.eg.db) %&gt;%
  dplyr::rename(ens_gene=ENSEMBL) %&gt;% filter(!duplicated(ens_gene)) %&gt;%
  inner_join(tblks %&gt;% dplyr::select(ens_gene,cluster,log2FoldChange),by=&quot;ens_gene&quot;) %&gt;%
  arrange(-log2FoldChange)</code></pre>
<pre><code>## &#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in bitr(tblks$ens_gene, fromType = &quot;ENSEMBL&quot;, toType =
## &quot;ENTREZID&quot;, : 3.14% of input gene IDs are fail to map...</code></pre>
<pre class="r"><code>v &lt;- gsks$log2FoldChange
names(v) &lt;- gsks$ENTREZID
gse &lt;- gseGO(v,OrgDb = org.Mm.eg.db,pvalueCutoff = 0.05, nPerm=10000)</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<pre class="r"><code>gse@result[,-11] %&gt;% arrange(NES) %&gt;% tablemd </code></pre>
<table style="width:100%;">
<colgroup>
<col width="5%" />
<col width="42%" />
<col width="4%" />
<col width="8%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="2%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">ID</th>
<th align="left">Description</th>
<th align="right">setSize</th>
<th align="right">enrichmentScore</th>
<th align="right">NES</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalues</th>
<th align="right">rank</th>
<th align="left">leading_edge</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0042254" class="uri">GO:0042254</a></td>
<td align="left">ribosome biogenesis</td>
<td align="right">96</td>
<td align="right">-0.4634017</td>
<td align="right">-2.680988</td>
<td align="right">0.0001509</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">1257</td>
<td align="left">tags=68%, list=37%, signal=44%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0042255" class="uri">GO:0042255</a></td>
<td align="left">ribosome assembly</td>
<td align="right">24</td>
<td align="right">-0.6629834</td>
<td align="right">-2.659473</td>
<td align="right">0.0001700</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">561</td>
<td align="left">tags=71%, list=17%, signal=60%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006364" class="uri">GO:0006364</a></td>
<td align="left">rRNA processing</td>
<td align="right">73</td>
<td align="right">-0.4872374</td>
<td align="right">-2.649862</td>
<td align="right">0.0001562</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">1257</td>
<td align="left">tags=70%, list=37%, signal=45%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0006412" class="uri">GO:0006412</a></td>
<td align="left">translation</td>
<td align="right">178</td>
<td align="right">-0.4063480</td>
<td align="right">-2.638340</td>
<td align="right">0.0001412</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">636</td>
<td align="left">tags=38%, list=19%, signal=32%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0043043" class="uri">GO:0043043</a></td>
<td align="left">peptide biosynthetic process</td>
<td align="right">180</td>
<td align="right">-0.3995491</td>
<td align="right">-2.598825</td>
<td align="right">0.0001409</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">636</td>
<td align="left">tags=37%, list=19%, signal=32%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0016072" class="uri">GO:0016072</a></td>
<td align="left">rRNA metabolic process</td>
<td align="right">74</td>
<td align="right">-0.4737689</td>
<td align="right">-2.585301</td>
<td align="right">0.0001553</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">1257</td>
<td align="left">tags=69%, list=37%, signal=44%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0043604" class="uri">GO:0043604</a></td>
<td align="left">amide biosynthetic process</td>
<td align="right">188</td>
<td align="right">-0.3924679</td>
<td align="right">-2.574342</td>
<td align="right">0.0001388</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">636</td>
<td align="left">tags=37%, list=19%, signal=32%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0000028" class="uri">GO:0000028</a></td>
<td align="left">ribosomal small subunit assembly</td>
<td align="right">14</td>
<td align="right">-0.7568006</td>
<td align="right">-2.539865</td>
<td align="right">0.0001779</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">561</td>
<td align="left">tags=79%, list=17%, signal=66%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0042274" class="uri">GO:0042274</a></td>
<td align="left">ribosomal small subunit biogenesis</td>
<td align="right">29</td>
<td align="right">-0.5990087</td>
<td align="right">-2.537913</td>
<td align="right">0.0001687</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">561</td>
<td align="left">tags=52%, list=17%, signal=44%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0034470" class="uri">GO:0034470</a></td>
<td align="left">ncRNA processing</td>
<td align="right">114</td>
<td align="right">-0.4179494</td>
<td align="right">-2.505995</td>
<td align="right">0.0001471</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">1257</td>
<td align="left">tags=63%, list=37%, signal=41%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0042273" class="uri">GO:0042273</a></td>
<td align="left">ribosomal large subunit biogenesis</td>
<td align="right">32</td>
<td align="right">-0.5689274</td>
<td align="right">-2.494149</td>
<td align="right">0.0001663</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">1257</td>
<td align="left">tags=94%, list=37%, signal=60%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1901566" class="uri">GO:1901566</a></td>
<td align="left">organonitrogen compound biosynthetic process</td>
<td align="right">275</td>
<td align="right">-0.3510510</td>
<td align="right">-2.445954</td>
<td align="right">0.0001322</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">393</td>
<td align="left">tags=25%, list=12%, signal=24%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006518" class="uri">GO:0006518</a></td>
<td align="left">peptide metabolic process</td>
<td align="right">200</td>
<td align="right">-0.3649848</td>
<td align="right">-2.420368</td>
<td align="right">0.0001380</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">636</td>
<td align="left">tags=34%, list=19%, signal=30%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0035914" class="uri">GO:0035914</a></td>
<td align="left">skeletal muscle cell differentiation</td>
<td align="right">21</td>
<td align="right">-0.6247482</td>
<td align="right">-2.405639</td>
<td align="right">0.0001710</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">426</td>
<td align="left">tags=52%, list=13%, signal=46%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0010830" class="uri">GO:0010830</a></td>
<td align="left">regulation of myotube differentiation</td>
<td align="right">11</td>
<td align="right">-0.7639972</td>
<td align="right">-2.370700</td>
<td align="right">0.0003595</td>
<td align="right">0.0236213</td>
<td align="right">0.0213256</td>
<td align="right">565</td>
<td align="left">tags=82%, list=17%, signal=68%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0034660" class="uri">GO:0034660</a></td>
<td align="left">ncRNA metabolic process</td>
<td align="right">137</td>
<td align="right">-0.3813658</td>
<td align="right">-2.364902</td>
<td align="right">0.0001445</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">1257</td>
<td align="left">tags=63%, list=37%, signal=41%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0043603" class="uri">GO:0043603</a></td>
<td align="left">cellular amide metabolic process</td>
<td align="right">215</td>
<td align="right">-0.3489399</td>
<td align="right">-2.341816</td>
<td align="right">0.0001363</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">636</td>
<td align="left">tags=33%, list=19%, signal=29%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0014902" class="uri">GO:0014902</a></td>
<td align="left">myotube differentiation</td>
<td align="right">25</td>
<td align="right">-0.5468010</td>
<td align="right">-2.222237</td>
<td align="right">0.0006766</td>
<td align="right">0.0306144</td>
<td align="right">0.0276391</td>
<td align="right">426</td>
<td align="left">tags=48%, list=13%, signal=42%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0022613" class="uri">GO:0022613</a></td>
<td align="left">ribonucleoprotein complex biogenesis</td>
<td align="right">140</td>
<td align="right">-0.3501383</td>
<td align="right">-2.180401</td>
<td align="right">0.0001439</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">561</td>
<td align="left">tags=29%, list=17%, signal=25%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0002181" class="uri">GO:0002181</a></td>
<td align="left">cytoplasmic translation</td>
<td align="right">20</td>
<td align="right">-0.5763633</td>
<td align="right">-2.179164</td>
<td align="right">0.0010350</td>
<td align="right">0.0381232</td>
<td align="right">0.0344181</td>
<td align="right">294</td>
<td align="left">tags=40%, list=9%, signal=37%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0000470" class="uri">GO:0000470</a></td>
<td align="left">maturation of LSU-rRNA</td>
<td align="right">15</td>
<td align="right">-0.6328402</td>
<td align="right">-2.169581</td>
<td align="right">0.0010619</td>
<td align="right">0.0381842</td>
<td align="right">0.0344732</td>
<td align="right">1257</td>
<td align="left">tags=100%, list=37%, signal=63%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1901564" class="uri">GO:1901564</a></td>
<td align="left">organonitrogen compound metabolic process</td>
<td align="right">383</td>
<td align="right">-0.2985497</td>
<td align="right">-2.167152</td>
<td align="right">0.0001263</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">294</td>
<td align="left">tags=17%, list=9%, signal=17%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0007519" class="uri">GO:0007519</a></td>
<td align="left">skeletal muscle tissue development</td>
<td align="right">44</td>
<td align="right">-0.4426557</td>
<td align="right">-2.117838</td>
<td align="right">0.0006561</td>
<td align="right">0.0306144</td>
<td align="right">0.0276391</td>
<td align="right">474</td>
<td align="left">tags=36%, list=14%, signal=32%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0060538" class="uri">GO:0060538</a></td>
<td align="left">skeletal muscle organ development</td>
<td align="right">44</td>
<td align="right">-0.4426557</td>
<td align="right">-2.117838</td>
<td align="right">0.0006561</td>
<td align="right">0.0306144</td>
<td align="right">0.0276391</td>
<td align="right">474</td>
<td align="left">tags=36%, list=14%, signal=32%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006457" class="uri">GO:0006457</a></td>
<td align="left">protein folding</td>
<td align="right">54</td>
<td align="right">-0.3962633</td>
<td align="right">-2.007900</td>
<td align="right">0.0009547</td>
<td align="right">0.0362619</td>
<td align="right">0.0327377</td>
<td align="right">1225</td>
<td align="left">tags=65%, list=36%, signal=42%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0071826" class="uri">GO:0071826</a></td>
<td align="left">ribonucleoprotein complex subunit organization</td>
<td align="right">73</td>
<td align="right">-0.3588635</td>
<td align="right">-1.951695</td>
<td align="right">0.0007810</td>
<td align="right">0.0316438</td>
<td align="right">0.0285684</td>
<td align="right">561</td>
<td align="left">tags=34%, list=17%, signal=29%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0022618" class="uri">GO:0022618</a></td>
<td align="left">ribonucleoprotein complex assembly</td>
<td align="right">71</td>
<td align="right">-0.3551235</td>
<td align="right">-1.918422</td>
<td align="right">0.0012518</td>
<td align="right">0.0416853</td>
<td align="right">0.0376340</td>
<td align="right">561</td>
<td align="left">tags=34%, list=17%, signal=29%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0006396" class="uri">GO:0006396</a></td>
<td align="left">RNA processing</td>
<td align="right">262</td>
<td align="right">-0.2581398</td>
<td align="right">-1.784551</td>
<td align="right">0.0001331</td>
<td align="right">0.0227664</td>
<td align="right">0.0205538</td>
<td align="right">1268</td>
<td align="left">tags=48%, list=37%, signal=33%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0048870" class="uri">GO:0048870</a></td>
<td align="left">cell motility</td>
<td align="right">239</td>
<td align="right">0.2075884</td>
<td align="right">1.606753</td>
<td align="right">0.0011696</td>
<td align="right">0.0400461</td>
<td align="right">0.0361541</td>
<td align="right">249</td>
<td align="left">tags=9%, list=7%, signal=9%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0051674" class="uri">GO:0051674</a></td>
<td align="left">localization of cell</td>
<td align="right">239</td>
<td align="right">0.2075884</td>
<td align="right">1.606753</td>
<td align="right">0.0011696</td>
<td align="right">0.0400461</td>
<td align="right">0.0361541</td>
<td align="right">249</td>
<td align="left">tags=9%, list=7%, signal=9%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0030030" class="uri">GO:0030030</a></td>
<td align="left">cell projection organization</td>
<td align="right">286</td>
<td align="right">0.2099512</td>
<td align="right">1.676238</td>
<td align="right">0.0004179</td>
<td align="right">0.0241876</td>
<td align="right">0.0218369</td>
<td align="right">197</td>
<td align="left">tags=6%, list=6%, signal=6%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0016477" class="uri">GO:0016477</a></td>
<td align="left">cell migration</td>
<td align="right">227</td>
<td align="right">0.2181837</td>
<td align="right">1.678227</td>
<td align="right">0.0007628</td>
<td align="right">0.0314290</td>
<td align="right">0.0283745</td>
<td align="right">249</td>
<td align="left">tags=9%, list=7%, signal=9%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0002682" class="uri">GO:0002682</a></td>
<td align="left">regulation of immune system process</td>
<td align="right">185</td>
<td align="right">0.2264194</td>
<td align="right">1.680976</td>
<td align="right">0.0014050</td>
<td align="right">0.0455403</td>
<td align="right">0.0411144</td>
<td align="right">247</td>
<td align="left">tags=11%, list=7%, signal=11%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0006928" class="uri">GO:0006928</a></td>
<td align="left">movement of cell or subcellular component</td>
<td align="right">308</td>
<td align="right">0.2104081</td>
<td align="right">1.702691</td>
<td align="right">0.0004281</td>
<td align="right">0.0242016</td>
<td align="right">0.0218495</td>
<td align="right">125</td>
<td align="left">tags=5%, list=4%, signal=5%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0051270" class="uri">GO:0051270</a></td>
<td align="left">regulation of cellular component movement</td>
<td align="right">161</td>
<td align="right">0.2419354</td>
<td align="right">1.751687</td>
<td align="right">0.0006800</td>
<td align="right">0.0306144</td>
<td align="right">0.0276391</td>
<td align="right">249</td>
<td align="left">tags=9%, list=7%, signal=8%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0006952" class="uri">GO:0006952</a></td>
<td align="left">defense response</td>
<td align="right">163</td>
<td align="right">0.2446053</td>
<td align="right">1.774868</td>
<td align="right">0.0006796</td>
<td align="right">0.0306144</td>
<td align="right">0.0276391</td>
<td align="right">366</td>
<td align="left">tags=18%, list=11%, signal=17%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0040012" class="uri">GO:0040012</a></td>
<td align="left">regulation of locomotion</td>
<td align="right">152</td>
<td align="right">0.2499171</td>
<td align="right">1.785794</td>
<td align="right">0.0003343</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">249</td>
<td align="left">tags=10%, list=7%, signal=10%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:2000145" class="uri">GO:2000145</a></td>
<td align="left">regulation of cell motility</td>
<td align="right">147</td>
<td align="right">0.2523772</td>
<td align="right">1.793686</td>
<td align="right">0.0006645</td>
<td align="right">0.0306144</td>
<td align="right">0.0276391</td>
<td align="right">296</td>
<td align="left">tags=10%, list=9%, signal=10%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0009653" class="uri">GO:0009653</a></td>
<td align="left">anatomical structure morphogenesis</td>
<td align="right">494</td>
<td align="right">0.2096843</td>
<td align="right">1.796830</td>
<td align="right">0.0005461</td>
<td align="right">0.0288628</td>
<td align="right">0.0260577</td>
<td align="right">33</td>
<td align="left">tags=1%, list=1%, signal=2%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0048646" class="uri">GO:0048646</a></td>
<td align="left">anatomical structure formation involved in morphogenesis</td>
<td align="right">234</td>
<td align="right">0.2328384</td>
<td align="right">1.796831</td>
<td align="right">0.0003876</td>
<td align="right">0.0239760</td>
<td align="right">0.0216458</td>
<td align="right">194</td>
<td align="left">tags=7%, list=6%, signal=7%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0072358" class="uri">GO:0072358</a></td>
<td align="left">cardiovascular system development</td>
<td align="right">133</td>
<td align="right">0.2596391</td>
<td align="right">1.802219</td>
<td align="right">0.0009794</td>
<td align="right">0.0366308</td>
<td align="right">0.0330707</td>
<td align="right">231</td>
<td align="left">tags=9%, list=7%, signal=9%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0040011" class="uri">GO:0040011</a></td>
<td align="left">locomotion</td>
<td align="right">263</td>
<td align="right">0.2298889</td>
<td align="right">1.809980</td>
<td align="right">0.0004044</td>
<td align="right">0.0239760</td>
<td align="right">0.0216458</td>
<td align="right">249</td>
<td align="left">tags=9%, list=7%, signal=9%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0007167" class="uri">GO:0007167</a></td>
<td align="left">enzyme linked receptor protein signaling pathway</td>
<td align="right">163</td>
<td align="right">0.2499473</td>
<td align="right">1.813629</td>
<td align="right">0.0003398</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">329</td>
<td align="left">tags=10%, list=10%, signal=10%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0001944" class="uri">GO:0001944</a></td>
<td align="left">vasculature development</td>
<td align="right">127</td>
<td align="right">0.2630866</td>
<td align="right">1.813897</td>
<td align="right">0.0009545</td>
<td align="right">0.0362619</td>
<td align="right">0.0327377</td>
<td align="right">231</td>
<td align="left">tags=9%, list=7%, signal=9%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0030334" class="uri">GO:0030334</a></td>
<td align="left">regulation of cell migration</td>
<td align="right">142</td>
<td align="right">0.2595409</td>
<td align="right">1.827920</td>
<td align="right">0.0003288</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">296</td>
<td align="left">tags=11%, list=9%, signal=10%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0032989" class="uri">GO:0032989</a></td>
<td align="left">cellular component morphogenesis</td>
<td align="right">260</td>
<td align="right">0.2369479</td>
<td align="right">1.864277</td>
<td align="right">0.0004018</td>
<td align="right">0.0239760</td>
<td align="right">0.0216458</td>
<td align="right">125</td>
<td align="left">tags=5%, list=4%, signal=5%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0001568" class="uri">GO:0001568</a></td>
<td align="left">blood vessel development</td>
<td align="right">118</td>
<td align="right">0.2807041</td>
<td align="right">1.904827</td>
<td align="right">0.0003112</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">231</td>
<td align="left">tags=9%, list=7%, signal=9%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0060560" class="uri">GO:0060560</a></td>
<td align="left">developmental growth involved in morphogenesis</td>
<td align="right">51</td>
<td align="right">0.3540667</td>
<td align="right">1.920758</td>
<td align="right">0.0010681</td>
<td align="right">0.0381842</td>
<td align="right">0.0344732</td>
<td align="right">508</td>
<td align="left">tags=33%, list=15%, signal=29%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0000902" class="uri">GO:0000902</a></td>
<td align="left">cell morphogenesis</td>
<td align="right">240</td>
<td align="right">0.2484828</td>
<td align="right">1.923769</td>
<td align="right">0.0003920</td>
<td align="right">0.0239760</td>
<td align="right">0.0216458</td>
<td align="right">189</td>
<td align="left">tags=6%, list=6%, signal=6%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0048514" class="uri">GO:0048514</a></td>
<td align="left">blood vessel morphogenesis</td>
<td align="right">96</td>
<td align="right">0.2985092</td>
<td align="right">1.934272</td>
<td align="right">0.0005926</td>
<td align="right">0.0300123</td>
<td align="right">0.0270955</td>
<td align="right">816</td>
<td align="left">tags=45%, list=24%, signal=35%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0007411" class="uri">GO:0007411</a></td>
<td align="left">axon guidance</td>
<td align="right">32</td>
<td align="right">0.4142017</td>
<td align="right">1.974554</td>
<td align="right">0.0007524</td>
<td align="right">0.0314290</td>
<td align="right">0.0283745</td>
<td align="right">688</td>
<td align="left">tags=56%, list=20%, signal=45%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0097485" class="uri">GO:0097485</a></td>
<td align="left">neuron projection guidance</td>
<td align="right">32</td>
<td align="right">0.4142017</td>
<td align="right">1.974554</td>
<td align="right">0.0007524</td>
<td align="right">0.0314290</td>
<td align="right">0.0283745</td>
<td align="right">688</td>
<td align="left">tags=56%, list=20%, signal=45%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0002831" class="uri">GO:0002831</a></td>
<td align="left">regulation of response to biotic stimulus</td>
<td align="right">38</td>
<td align="right">0.3938542</td>
<td align="right">1.981667</td>
<td align="right">0.0015236</td>
<td align="right">0.0474860</td>
<td align="right">0.0428710</td>
<td align="right">218</td>
<td align="left">tags=26%, list=6%, signal=25%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0001654" class="uri">GO:0001654</a></td>
<td align="left">eye development</td>
<td align="right">80</td>
<td align="right">0.3201580</td>
<td align="right">1.985146</td>
<td align="right">0.0005708</td>
<td align="right">0.0295225</td>
<td align="right">0.0266533</td>
<td align="right">968</td>
<td align="left">tags=52%, list=29%, signal=38%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0030031" class="uri">GO:0030031</a></td>
<td align="left">cell projection assembly</td>
<td align="right">95</td>
<td align="right">0.3082206</td>
<td align="right">1.995411</td>
<td align="right">0.0002948</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">968</td>
<td align="left">tags=49%, list=29%, signal=36%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0007179" class="uri">GO:0007179</a></td>
<td align="left">transforming growth factor beta receptor signaling pathway</td>
<td align="right">36</td>
<td align="right">0.4056002</td>
<td align="right">2.008732</td>
<td align="right">0.0005075</td>
<td align="right">0.0274155</td>
<td align="right">0.0247510</td>
<td align="right">453</td>
<td align="left">tags=36%, list=13%, signal=32%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0050808" class="uri">GO:0050808</a></td>
<td align="left">synapse organization</td>
<td align="right">45</td>
<td align="right">0.3855047</td>
<td align="right">2.030472</td>
<td align="right">0.0012867</td>
<td align="right">0.0422689</td>
<td align="right">0.0381608</td>
<td align="right">881</td>
<td align="left">tags=53%, list=26%, signal=40%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0090092" class="uri">GO:0090092</a></td>
<td align="left">regulation of transmembrane receptor protein serine/threonine kinase signaling pathway</td>
<td align="right">38</td>
<td align="right">0.4065839</td>
<td align="right">2.045717</td>
<td align="right">0.0007618</td>
<td align="right">0.0314290</td>
<td align="right">0.0283745</td>
<td align="right">740</td>
<td align="left">tags=47%, list=22%, signal=37%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0032990" class="uri">GO:0032990</a></td>
<td align="left">cell part morphogenesis</td>
<td align="right">182</td>
<td align="right">0.2770484</td>
<td align="right">2.046520</td>
<td align="right">0.0003516</td>
<td align="right">0.0236213</td>
<td align="right">0.0213256</td>
<td align="right">125</td>
<td align="left">tags=4%, list=4%, signal=4%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0002252" class="uri">GO:0002252</a></td>
<td align="left">immune effector process</td>
<td align="right">106</td>
<td align="right">0.3160199</td>
<td align="right">2.091120</td>
<td align="right">0.0003048</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">535</td>
<td align="left">tags=31%, list=16%, signal=27%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0017015" class="uri">GO:0017015</a></td>
<td align="left">regulation of transforming growth factor beta receptor signaling pathway</td>
<td align="right">23</td>
<td align="right">0.4905627</td>
<td align="right">2.097185</td>
<td align="right">0.0014507</td>
<td align="right">0.0457999</td>
<td align="right">0.0413488</td>
<td align="right">549</td>
<td align="left">tags=48%, list=16%, signal=40%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1903844" class="uri">GO:1903844</a></td>
<td align="left">regulation of cellular response to transforming growth factor beta stimulus</td>
<td align="right">23</td>
<td align="right">0.4905627</td>
<td align="right">2.097185</td>
<td align="right">0.0014507</td>
<td align="right">0.0457999</td>
<td align="right">0.0413488</td>
<td align="right">549</td>
<td align="left">tags=48%, list=16%, signal=40%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0019724" class="uri">GO:0019724</a></td>
<td align="left">B cell mediated immunity</td>
<td align="right">14</td>
<td align="right">0.5847689</td>
<td align="right">2.103819</td>
<td align="right">0.0011410</td>
<td align="right">0.0400461</td>
<td align="right">0.0361541</td>
<td align="right">175</td>
<td align="left">tags=43%, list=5%, signal=41%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0098792" class="uri">GO:0098792</a></td>
<td align="left">xenophagy</td>
<td align="right">23</td>
<td align="right">0.5043330</td>
<td align="right">2.156054</td>
<td align="right">0.0012089</td>
<td align="right">0.0408171</td>
<td align="right">0.0368502</td>
<td align="right">149</td>
<td align="left">tags=30%, list=4%, signal=29%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0048858" class="uri">GO:0048858</a></td>
<td align="left">cell projection morphogenesis</td>
<td align="right">173</td>
<td align="right">0.2966836</td>
<td align="right">2.178018</td>
<td align="right">0.0003391</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">125</td>
<td align="left">tags=5%, list=4%, signal=5%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0050691" class="uri">GO:0050691</a></td>
<td align="left">regulation of defense response to virus by host</td>
<td align="right">27</td>
<td align="right">0.4873594</td>
<td align="right">2.196338</td>
<td align="right">0.0002456</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">149</td>
<td align="left">tags=30%, list=4%, signal=29%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0090288" class="uri">GO:0090288</a></td>
<td align="left">negative regulation of cellular response to growth factor stimulus</td>
<td align="right">27</td>
<td align="right">0.4888537</td>
<td align="right">2.203072</td>
<td align="right">0.0002456</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">910</td>
<td align="left">tags=63%, list=27%, signal=46%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1901343" class="uri">GO:1901343</a></td>
<td align="left">negative regulation of vasculature development</td>
<td align="right">18</td>
<td align="right">0.5616572</td>
<td align="right">2.206256</td>
<td align="right">0.0007034</td>
<td align="right">0.0310903</td>
<td align="right">0.0280687</td>
<td align="right">810</td>
<td align="left">tags=67%, list=24%, signal=51%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0051928" class="uri">GO:0051928</a></td>
<td align="left">positive regulation of calcium ion transport</td>
<td align="right">19</td>
<td align="right">0.5520963</td>
<td align="right">2.213509</td>
<td align="right">0.0004687</td>
<td align="right">0.0258964</td>
<td align="right">0.0233796</td>
<td align="right">779</td>
<td align="left">tags=68%, list=23%, signal=53%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0030512" class="uri">GO:0030512</a></td>
<td align="left">negative regulation of transforming growth factor beta receptor signaling pathway</td>
<td align="right">16</td>
<td align="right">0.5946019</td>
<td align="right">2.242672</td>
<td align="right">0.0009255</td>
<td align="right">0.0362619</td>
<td align="right">0.0327377</td>
<td align="right">421</td>
<td align="left">tags=56%, list=12%, signal=50%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1903845" class="uri">GO:1903845</a></td>
<td align="left">negative regulation of cellular response to transforming growth factor beta stimulus</td>
<td align="right">16</td>
<td align="right">0.5946019</td>
<td align="right">2.242672</td>
<td align="right">0.0009255</td>
<td align="right">0.0362619</td>
<td align="right">0.0327377</td>
<td align="right">421</td>
<td align="left">tags=56%, list=12%, signal=50%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0010927" class="uri">GO:0010927</a></td>
<td align="left">cellular component assembly involved in morphogenesis</td>
<td align="right">73</td>
<td align="right">0.3704136</td>
<td align="right">2.244189</td>
<td align="right">0.0002778</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">889</td>
<td align="left">tags=56%, list=26%, signal=42%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0050688" class="uri">GO:0050688</a></td>
<td align="left">regulation of defense response to virus</td>
<td align="right">29</td>
<td align="right">0.4895229</td>
<td align="right">2.264761</td>
<td align="right">0.0002454</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">170</td>
<td align="left">tags=31%, list=5%, signal=30%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0002230" class="uri">GO:0002230</a></td>
<td align="left">positive regulation of defense response to virus by host</td>
<td align="right">24</td>
<td align="right">0.5275932</td>
<td align="right">2.290373</td>
<td align="right">0.0002428</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">149</td>
<td align="left">tags=33%, list=4%, signal=32%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0044782" class="uri">GO:0044782</a></td>
<td align="left">cilium organization</td>
<td align="right">65</td>
<td align="right">0.3919042</td>
<td align="right">2.290703</td>
<td align="right">0.0002751</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">1423</td>
<td align="left">tags=77%, list=42%, signal=46%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0060271" class="uri">GO:0060271</a></td>
<td align="left">cilium morphogenesis</td>
<td align="right">65</td>
<td align="right">0.3968352</td>
<td align="right">2.319525</td>
<td align="right">0.0002751</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">1434</td>
<td align="left">tags=77%, list=42%, signal=45%</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0042384" class="uri">GO:0042384</a></td>
<td align="left">cilium assembly</td>
<td align="right">57</td>
<td align="right">0.4335889</td>
<td align="right">2.444957</td>
<td align="right">0.0002692</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">1371</td>
<td align="left">tags=79%, list=40%, signal=48%</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0048536" class="uri">GO:0048536</a></td>
<td align="left">spleen development</td>
<td align="right">11</td>
<td align="right">0.7795684</td>
<td align="right">2.546111</td>
<td align="right">0.0002253</td>
<td align="right">0.0236008</td>
<td align="right">0.0213071</td>
<td align="right">622</td>
<td align="left">tags=91%, list=18%, signal=74%</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="nih3t3-h3voe" class="section level2">
<h2>NIH3T3 H3vOE</h2>
<pre class="r"><code>defn &lt;- getdef(&quot;data/table_3T3RNAseq.txt&quot;)
ddsn &lt;- getdds(&quot;data/countGENCODEvM9_3T3RNAseq.edit.txt&quot;,defn,~time+cond)</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre class="r"><code>rldn &lt;- rlog(ddsn,blind=FALSE)
tablemd(colData(ddsn))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">cond</th>
<th align="left">time</th>
<th align="left">rep</th>
<th align="left">clone</th>
<th align="right">sizeFactor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">X3T3.WT.noinf</td>
<td align="left">WT</td>
<td align="left">noinf</td>
<td align="left">Rep1</td>
<td align="left">WT_Rep1</td>
<td align="right">1.0082323</td>
</tr>
<tr class="even">
<td align="left">X3T3.WT.infect</td>
<td align="left">WT</td>
<td align="left">infect</td>
<td align="left">Rep1</td>
<td align="left">WT_Rep1</td>
<td align="right">1.2420505</td>
</tr>
<tr class="odd">
<td align="left">X3T3.WT.inf24</td>
<td align="left">WT</td>
<td align="left">inf24</td>
<td align="left">Rep1</td>
<td align="left">WT_Rep1</td>
<td align="right">0.9808936</td>
</tr>
<tr class="even">
<td align="left">X3T3.WT.inf48</td>
<td align="left">WT</td>
<td align="left">inf48</td>
<td align="left">Rep1</td>
<td align="left">WT_Rep1</td>
<td align="right">0.9110183</td>
</tr>
<tr class="odd">
<td align="left">X3T3.H3v3.noinf</td>
<td align="left">H3v3</td>
<td align="left">noinf</td>
<td align="left">Rep1</td>
<td align="left">H3v3_Rep1</td>
<td align="right">1.1716920</td>
</tr>
<tr class="even">
<td align="left">X3T3.H3v3.infect</td>
<td align="left">H3v3</td>
<td align="left">infect</td>
<td align="left">Rep1</td>
<td align="left">H3v3_Rep1</td>
<td align="right">1.0533892</td>
</tr>
<tr class="odd">
<td align="left">X3T3.H3v3.inf24</td>
<td align="left">H3v3</td>
<td align="left">inf24</td>
<td align="left">Rep1</td>
<td align="left">H3v3_Rep1</td>
<td align="right">0.9724215</td>
</tr>
<tr class="even">
<td align="left">X3T3.H3v3.inf48</td>
<td align="left">H3v3</td>
<td align="left">inf48</td>
<td align="left">Rep1</td>
<td align="left">H3v3_Rep1</td>
<td align="right">0.8092273</td>
</tr>
<tr class="odd">
<td align="left">X3T3.H3v7.noinf</td>
<td align="left">H3v7</td>
<td align="left">noinf</td>
<td align="left">Rep1</td>
<td align="left">H3v7_Rep1</td>
<td align="right">1.1884297</td>
</tr>
<tr class="even">
<td align="left">X3T3.H3v7.infect</td>
<td align="left">H3v7</td>
<td align="left">infect</td>
<td align="left">Rep1</td>
<td align="left">H3v7_Rep1</td>
<td align="right">1.1936027</td>
</tr>
<tr class="odd">
<td align="left">X3T3.H3v7.inf24</td>
<td align="left">H3v7</td>
<td align="left">inf24</td>
<td align="left">Rep1</td>
<td align="left">H3v7_Rep1</td>
<td align="right">0.9042245</td>
</tr>
<tr class="even">
<td align="left">X3T3.H3v7.inf48</td>
<td align="left">H3v7</td>
<td align="left">inf48</td>
<td align="left">Rep1</td>
<td align="left">H3v7_Rep1</td>
<td align="right">0.8198526</td>
</tr>
</tbody>
</table>
<pre class="r"><code>set.seed(333)
resn &lt;- results(ddsn,contrast=c(&quot;cond&quot;,&quot;H3v7&quot;,&quot;WT&quot;))
tbln &lt;- timecoursePlots(resn,rldn,k=6,showsd=TRUE)</code></pre>
<pre><code>## Warning in lda.default(x, grouping, ...): variables are collinear</code></pre>
<p><img src="RNA-seq_files/figure-html/results_NIH3T3-1.png" /><!-- --><img src="RNA-seq_files/figure-html/results_NIH3T3-2.png" /><!-- --><img src="RNA-seq_files/figure-html/results_NIH3T3-3.png" /><!-- --></p>
<pre class="r"><code>p &lt;- ggplot(tbln,aes(cluster,baseMean,fill=cluster)) + 
  geom_violin() + scale_y_log10() + theme_bw() +
  scale_fill_brewer(palette=&quot;Set2&quot;)
print(p)</code></pre>
<p><img src="RNA-seq_files/figure-html/results_NIH3T3-4.png" /><!-- --></p>
<pre class="r"><code>B &lt;- with(tbln %&gt;% modTbl, table(biotype,cluster))
pal &lt;- brewer.pal(6,&quot;Set1&quot;)
op &lt;- par(mai=c(.5,2.2,.5,.5))
barplot(t(B[order(-rowSums(B))[1:6],]),beside=TRUE,col=pal,legend.text = colnames(B),las=1,horiz = TRUE)</code></pre>
<p><img src="RNA-seq_files/figure-html/results_NIH3T3-5.png" /><!-- --></p>
<pre class="r"><code>par(op)
#showDT(tbln)</code></pre>
<pre class="r"><code>set.seed(333)
tblns &lt;- timecoursePlots(resn[resn$baseMean&gt;100,],rldn,k=6)</code></pre>
<pre><code>## Warning in lda.default(x, grouping, ...): variables are collinear</code></pre>
<p><img src="RNA-seq_files/figure-html/resultsNIH3T3Sel-1.png" /><!-- --><img src="RNA-seq_files/figure-html/resultsNIH3T3Sel-2.png" /><!-- --><img src="RNA-seq_files/figure-html/resultsNIH3T3Sel-3.png" /><!-- --></p>
<pre class="r"><code>#showDT(tblns)</code></pre>
<div id="gsea-1" class="section level4">
<h4>GSEA</h4>
<pre class="r"><code>gsksn &lt;- bitr(tblns$ens_gene,fromType = &quot;ENSEMBL&quot;,toType = &quot;ENTREZID&quot;,org.Mm.eg.db) %&gt;%
  dplyr::rename(ens_gene=ENSEMBL) %&gt;% filter(!duplicated(ens_gene)) %&gt;%
  inner_join(tblns %&gt;% dplyr::select(ens_gene,cluster,log2FoldChange),by=&quot;ens_gene&quot;) %&gt;%
  arrange(-log2FoldChange)</code></pre>
<pre><code>## &#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in bitr(tblns$ens_gene, fromType = &quot;ENSEMBL&quot;, toType =
## &quot;ENTREZID&quot;, : 2.85% of input gene IDs are fail to map...</code></pre>
<pre class="r"><code>v &lt;- with(gsksn,setNames(log2FoldChange,ens_gene)) %&gt;% sort(decreasing=TRUE)
gsen &lt;- gseGO(v,ont = &quot;BP&quot;,OrgDb = org.Mm.eg.db,pvalueCutoff = 0.25,nPerm=10000,keytype = &quot;ENSEMBL&quot;)</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<pre class="r"><code>tablemd(gsen@result[,-c(1,11)] %&gt;% arrange(desc(enrichmentScore)))</code></pre>
<table>
<colgroup>
<col width="30%" />
<col width="5%" />
<col width="10%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="3%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Description</th>
<th align="right">setSize</th>
<th align="right">enrichmentScore</th>
<th align="right">NES</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalues</th>
<th align="right">rank</th>
<th align="left">leading_edge</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">digestion</td>
<td align="right">13</td>
<td align="right">0.7825727</td>
<td align="right">2.414145</td>
<td align="right">0.0001931</td>
<td align="right">0.0534883</td>
<td align="right">0.0501514</td>
<td align="right">141</td>
<td align="left">tags=54%, list=5%, signal=52%</td>
</tr>
<tr class="even">
<td align="left">digestive system process</td>
<td align="right">12</td>
<td align="right">0.7639728</td>
<td align="right">2.303546</td>
<td align="right">0.0001948</td>
<td align="right">0.0534883</td>
<td align="right">0.0501514</td>
<td align="right">141</td>
<td align="left">tags=50%, list=5%, signal=48%</td>
</tr>
<tr class="odd">
<td align="left">negative regulation of calcium ion transport</td>
<td align="right">10</td>
<td align="right">0.7040384</td>
<td align="right">1.996329</td>
<td align="right">0.0025722</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">313</td>
<td align="left">tags=60%, list=10%, signal=54%</td>
</tr>
<tr class="even">
<td align="left">sensory perception of pain</td>
<td align="right">15</td>
<td align="right">0.6949529</td>
<td align="right">2.248935</td>
<td align="right">0.0001884</td>
<td align="right">0.0534883</td>
<td align="right">0.0501514</td>
<td align="right">202</td>
<td align="left">tags=53%, list=7%, signal=50%</td>
</tr>
<tr class="odd">
<td align="left">response to purine-containing compound</td>
<td align="right">11</td>
<td align="right">0.6937279</td>
<td align="right">2.036726</td>
<td align="right">0.0027232</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">350</td>
<td align="left">tags=64%, list=11%, signal=57%</td>
</tr>
<tr class="even">
<td align="left">response to organophosphorus</td>
<td align="right">11</td>
<td align="right">0.6937279</td>
<td align="right">2.036726</td>
<td align="right">0.0027232</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">350</td>
<td align="left">tags=64%, list=11%, signal=57%</td>
</tr>
<tr class="odd">
<td align="left">membrane depolarization</td>
<td align="right">19</td>
<td align="right">0.5772389</td>
<td align="right">2.007357</td>
<td align="right">0.0033470</td>
<td align="right">0.2456597</td>
<td align="right">0.2303337</td>
<td align="right">197</td>
<td align="left">tags=37%, list=6%, signal=35%</td>
</tr>
<tr class="even">
<td align="left">regulation of muscle contraction</td>
<td align="right">30</td>
<td align="right">0.5452362</td>
<td align="right">2.159552</td>
<td align="right">0.0005337</td>
<td align="right">0.1318804</td>
<td align="right">0.1236528</td>
<td align="right">364</td>
<td align="left">tags=57%, list=12%, signal=50%</td>
</tr>
<tr class="odd">
<td align="left">striated muscle contraction</td>
<td align="right">28</td>
<td align="right">0.5397559</td>
<td align="right">2.100692</td>
<td align="right">0.0012414</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">364</td>
<td align="left">tags=54%, list=12%, signal=48%</td>
</tr>
<tr class="even">
<td align="left">regulation of muscle tissue development</td>
<td align="right">27</td>
<td align="right">0.5370386</td>
<td align="right">2.068711</td>
<td align="right">0.0017864</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">375</td>
<td align="left">tags=41%, list=12%, signal=36%</td>
</tr>
<tr class="odd">
<td align="left">mononuclear cell proliferation</td>
<td align="right">47</td>
<td align="right">0.5034080</td>
<td align="right">2.252879</td>
<td align="right">0.0001706</td>
<td align="right">0.0534883</td>
<td align="right">0.0501514</td>
<td align="right">364</td>
<td align="left">tags=38%, list=12%, signal=34%</td>
</tr>
<tr class="even">
<td align="left">muscle cell proliferation</td>
<td align="right">29</td>
<td align="right">0.5030419</td>
<td align="right">1.976027</td>
<td align="right">0.0030174</td>
<td align="right">0.2405155</td>
<td align="right">0.2255105</td>
<td align="right">303</td>
<td align="left">tags=28%, list=10%, signal=25%</td>
</tr>
<tr class="odd">
<td align="left">T cell proliferation</td>
<td align="right">30</td>
<td align="right">0.5015888</td>
<td align="right">1.986675</td>
<td align="right">0.0033802</td>
<td align="right">0.2456597</td>
<td align="right">0.2303337</td>
<td align="right">466</td>
<td align="left">tags=40%, list=15%, signal=34%</td>
</tr>
<tr class="even">
<td align="left">lymphocyte proliferation</td>
<td align="right">45</td>
<td align="right">0.4951832</td>
<td align="right">2.189229</td>
<td align="right">0.0001719</td>
<td align="right">0.0534883</td>
<td align="right">0.0501514</td>
<td align="right">364</td>
<td align="left">tags=38%, list=12%, signal=34%</td>
</tr>
<tr class="odd">
<td align="left">response to drug</td>
<td align="right">49</td>
<td align="right">0.4947984</td>
<td align="right">2.243040</td>
<td align="right">0.0001691</td>
<td align="right">0.0534883</td>
<td align="right">0.0501514</td>
<td align="right">332</td>
<td align="left">tags=33%, list=11%, signal=30%</td>
</tr>
<tr class="even">
<td align="left">leukocyte proliferation</td>
<td align="right">48</td>
<td align="right">0.4937891</td>
<td align="right">2.226438</td>
<td align="right">0.0001692</td>
<td align="right">0.0534883</td>
<td align="right">0.0501514</td>
<td align="right">364</td>
<td align="left">tags=38%, list=12%, signal=34%</td>
</tr>
<tr class="odd">
<td align="left">regulation of mononuclear cell proliferation</td>
<td align="right">35</td>
<td align="right">0.4877971</td>
<td align="right">2.020568</td>
<td align="right">0.0022652</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">364</td>
<td align="left">tags=37%, list=12%, signal=33%</td>
</tr>
<tr class="even">
<td align="left">regulation of leukocyte proliferation</td>
<td align="right">36</td>
<td align="right">0.4756325</td>
<td align="right">1.983491</td>
<td align="right">0.0027821</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">364</td>
<td align="left">tags=36%, list=12%, signal=32%</td>
</tr>
<tr class="odd">
<td align="left">regulation of muscle system process</td>
<td align="right">41</td>
<td align="right">0.4641350</td>
<td align="right">2.002661</td>
<td align="right">0.0020708</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">364</td>
<td align="left">tags=44%, list=12%, signal=39%</td>
</tr>
<tr class="even">
<td align="left">muscle contraction</td>
<td align="right">49</td>
<td align="right">0.4286367</td>
<td align="right">1.943113</td>
<td align="right">0.0028750</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">364</td>
<td align="left">tags=43%, list=12%, signal=38%</td>
</tr>
<tr class="odd">
<td align="left">regulation of membrane potential</td>
<td align="right">58</td>
<td align="right">0.4189125</td>
<td align="right">1.977505</td>
<td align="right">0.0018257</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">380</td>
<td align="left">tags=31%, list=12%, signal=28%</td>
</tr>
<tr class="even">
<td align="left">regulation of system process</td>
<td align="right">79</td>
<td align="right">0.4146072</td>
<td align="right">2.095371</td>
<td align="right">0.0001614</td>
<td align="right">0.0534883</td>
<td align="right">0.0501514</td>
<td align="right">369</td>
<td align="left">tags=35%, list=12%, signal=32%</td>
</tr>
<tr class="odd">
<td align="left">regulation of homeostatic process</td>
<td align="right">95</td>
<td align="right">0.3400514</td>
<td align="right">1.785078</td>
<td align="right">0.0033014</td>
<td align="right">0.2456597</td>
<td align="right">0.2303337</td>
<td align="right">380</td>
<td align="left">tags=24%, list=12%, signal=22%</td>
</tr>
<tr class="even">
<td align="left">system process</td>
<td align="right">229</td>
<td align="right">0.3266303</td>
<td align="right">1.993322</td>
<td align="right">0.0001414</td>
<td align="right">0.0534883</td>
<td align="right">0.0501514</td>
<td align="right">15</td>
<td align="left">tags=1%, list=0%, signal=1%</td>
</tr>
<tr class="odd">
<td align="left">cell activation</td>
<td align="right">144</td>
<td align="right">0.3178392</td>
<td align="right">1.805196</td>
<td align="right">0.0013467</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">431</td>
<td align="left">tags=19%, list=14%, signal=17%</td>
</tr>
<tr class="even">
<td align="left">neurological system process</td>
<td align="right">131</td>
<td align="right">0.3157904</td>
<td align="right">1.762131</td>
<td align="right">0.0022800</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">15</td>
<td align="left">tags=2%, list=0%, signal=2%</td>
</tr>
<tr class="odd">
<td align="left">single organismal cell-cell adhesion</td>
<td align="right">132</td>
<td align="right">0.3123182</td>
<td align="right">1.744688</td>
<td align="right">0.0025844</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">273</td>
<td align="left">tags=13%, list=9%, signal=12%</td>
</tr>
<tr class="even">
<td align="left">single organism cell adhesion</td>
<td align="right">143</td>
<td align="right">0.3063934</td>
<td align="right">1.734911</td>
<td align="right">0.0024111</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">273</td>
<td align="left">tags=13%, list=9%, signal=12%</td>
</tr>
<tr class="odd">
<td align="left">cell adhesion</td>
<td align="right">284</td>
<td align="right">0.2662296</td>
<td align="right">1.678478</td>
<td align="right">0.0012290</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">194</td>
<td align="left">tags=8%, list=6%, signal=8%</td>
</tr>
<tr class="even">
<td align="left">biological adhesion</td>
<td align="right">286</td>
<td align="right">0.2628691</td>
<td align="right">1.658732</td>
<td align="right">0.0019105</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">76</td>
<td align="left">tags=8%, list=2%, signal=9%</td>
</tr>
<tr class="odd">
<td align="left">regulation of multicellular organismal process</td>
<td align="right">461</td>
<td align="right">0.2286622</td>
<td align="right">1.529620</td>
<td align="right">0.0029107</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">1</td>
<td align="left">tags=1%, list=0%, signal=1%</td>
</tr>
<tr class="even">
<td align="left">defense response</td>
<td align="right">236</td>
<td align="right">-0.2349582</td>
<td align="right">-1.639909</td>
<td align="right">0.0020848</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">435</td>
<td align="left">tags=27%, list=14%, signal=25%</td>
</tr>
<tr class="odd">
<td align="left">response to cytokine</td>
<td align="right">107</td>
<td align="right">-0.2906225</td>
<td align="right">-1.765069</td>
<td align="right">0.0016704</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">423</td>
<td align="left">tags=32%, list=14%, signal=28%</td>
</tr>
<tr class="even">
<td align="left">G-protein coupled receptor signaling pathway</td>
<td align="right">58</td>
<td align="right">-0.3522887</td>
<td align="right">-1.848661</td>
<td align="right">0.0027659</td>
<td align="right">0.2397410</td>
<td align="right">0.2247843</td>
<td align="right">141</td>
<td align="left">tags=26%, list=5%, signal=25%</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="si-mh2a.1-in-c2c12" class="section level2">
<h2>si-mH2A.1 in C2C12</h2>
<ul>
<li><a href="http://www.sciencedirect.com/science/article/pii/S2211124716000036" class="uri">http://www.sciencedirect.com/science/article/pii/S2211124716000036</a></li>
<li><a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP067391" class="uri">https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP067391</a></li>
</ul>
<pre class="r"><code>defm &lt;- getdef(&quot;data/table_mH2A1-2i.txt&quot;)
ddsm &lt;- getdds(&quot;data/countGENCODEvM9_C2C12-mH2A1-2i_RNAseq.edit.txt&quot;,defm,~time*cond,~time+cond)</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre class="r"><code>rldm &lt;- rlog(ddsm,blind=FALSE)
tablemd(colData(ddsm))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">cond</th>
<th align="left">time</th>
<th align="left">rep</th>
<th align="left">clone</th>
<th align="right">sizeFactor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SRR3000005</td>
<td align="left">CTL</td>
<td align="left">MB</td>
<td align="left">Rep1</td>
<td align="left">CTL_Rep1</td>
<td align="right">1.2282098</td>
</tr>
<tr class="even">
<td align="left">SRR3000006</td>
<td align="left">CTL</td>
<td align="left">MB</td>
<td align="left">Rep2</td>
<td align="left">CTL_Rep2</td>
<td align="right">1.1787304</td>
</tr>
<tr class="odd">
<td align="left">SRR3000007</td>
<td align="left">CTL</td>
<td align="left">MB</td>
<td align="left">Rep3</td>
<td align="left">CTL_Rep3</td>
<td align="right">2.2332695</td>
</tr>
<tr class="even">
<td align="left">SRR3000008</td>
<td align="left">CTL</td>
<td align="left">MT</td>
<td align="left">Rep1</td>
<td align="left">CTL_Rep1</td>
<td align="right">0.6890653</td>
</tr>
<tr class="odd">
<td align="left">SRR3000009</td>
<td align="left">CTL</td>
<td align="left">MT</td>
<td align="left">Rep2</td>
<td align="left">CTL_Rep2</td>
<td align="right">0.6294310</td>
</tr>
<tr class="even">
<td align="left">SRR3000010</td>
<td align="left">CTL</td>
<td align="left">MT</td>
<td align="left">Rep3</td>
<td align="left">CTL_Rep3</td>
<td align="right">0.5539811</td>
</tr>
<tr class="odd">
<td align="left">SRR3000011</td>
<td align="left">mH2A1.2i</td>
<td align="left">MB</td>
<td align="left">Rep1</td>
<td align="left">mH2A1.2i_Rep1</td>
<td align="right">0.9592084</td>
</tr>
<tr class="even">
<td align="left">SRR3000012</td>
<td align="left">mH2A1.2i</td>
<td align="left">MB</td>
<td align="left">Rep2</td>
<td align="left">mH2A1.2i_Rep2</td>
<td align="right">1.0371868</td>
</tr>
<tr class="odd">
<td align="left">SRR3000013</td>
<td align="left">mH2A1.2i</td>
<td align="left">MB</td>
<td align="left">Rep3</td>
<td align="left">mH2A1.2i_Rep3</td>
<td align="right">2.5351417</td>
</tr>
<tr class="even">
<td align="left">SRR3000014</td>
<td align="left">mH2A1.2i</td>
<td align="left">MT</td>
<td align="left">Rep1</td>
<td align="left">mH2A1.2i_Rep1</td>
<td align="right">0.9748941</td>
</tr>
<tr class="odd">
<td align="left">SRR3000015</td>
<td align="left">mH2A1.2i</td>
<td align="left">MT</td>
<td align="left">Rep2</td>
<td align="left">mH2A1.2i_Rep2</td>
<td align="right">0.7125510</td>
</tr>
<tr class="even">
<td align="left">SRR3000016</td>
<td align="left">mH2A1.2i</td>
<td align="left">MT</td>
<td align="left">Rep3</td>
<td align="left">mH2A1.2i_Rep3</td>
<td align="right">0.8126225</td>
</tr>
</tbody>
</table>
<pre class="r"><code>resm &lt;- results(ddsm,name=&quot;timeMT.condmH2A1.2i&quot;)
tblm &lt;- timecoursePlots(resm,rldm,k=4)</code></pre>
<pre><code>## Warning in lda.default(x, grouping, ...): variables are collinear</code></pre>
<p><img src="RNA-seq_files/figure-html/results_siMacroH2A1-1.png" /><!-- --><img src="RNA-seq_files/figure-html/results_siMacroH2A1-2.png" /><!-- --><img src="RNA-seq_files/figure-html/results_siMacroH2A1-3.png" /><!-- --></p>
<pre class="r"><code>#showDT(tblm)</code></pre>
</div>
<div id="bmal1-ko-in-hip-cartilage" class="section level2">
<h2>BMAL1-KO in hip cartilage</h2>
<ul>
<li><a href="https://www.jci.org/articles/view/82755" class="uri">https://www.jci.org/articles/view/82755</a></li>
<li><a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=ERP009958" class="uri">https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=ERP009958</a></li>
</ul>
<pre class="r"><code>defb &lt;- getdef(&quot;data/table_BMAL1KO.txt&quot;)
defb$cond &lt;- factor(defb$cond,c(&quot;WT&quot;,&quot;KO&quot;))
defb$time &lt;- factor(defb$time,c(&quot;5AM&quot;,&quot;9AM&quot;,&quot;5PM&quot;,&quot;9PM&quot;))
ddsb &lt;- getdds(&quot;data/countGENCODEvM9_cartilage_BMAL1-KO_RNAseq.edit.txt&quot;,defb,~time*cond,~time+cond)</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre class="r"><code>rldb &lt;- rlog(ddsb,blind=FALSE)
tablemd(colData(ddsb))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">cond</th>
<th align="left">time</th>
<th align="left">rep</th>
<th align="left">clone</th>
<th align="right">sizeFactor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ERR793472</td>
<td align="left">WT</td>
<td align="left">9AM</td>
<td align="left">Rep3</td>
<td align="left">WT_Rep3</td>
<td align="right">1.1692521</td>
</tr>
<tr class="even">
<td align="left">ERR793473</td>
<td align="left">KO</td>
<td align="left">5PM</td>
<td align="left">Rep3</td>
<td align="left">KO_Rep3</td>
<td align="right">1.1574875</td>
</tr>
<tr class="odd">
<td align="left">ERR793474</td>
<td align="left">WT</td>
<td align="left">9AM</td>
<td align="left">Rep2</td>
<td align="left">WT_Rep2</td>
<td align="right">1.7839916</td>
</tr>
<tr class="even">
<td align="left">ERR793475</td>
<td align="left">KO</td>
<td align="left">5AM</td>
<td align="left">Rep3</td>
<td align="left">KO_Rep3</td>
<td align="right">1.3333196</td>
</tr>
<tr class="odd">
<td align="left">ERR793476</td>
<td align="left">KO</td>
<td align="left">5AM</td>
<td align="left">Rep2</td>
<td align="left">KO_Rep2</td>
<td align="right">0.8373409</td>
</tr>
<tr class="even">
<td align="left">ERR793477</td>
<td align="left">WT</td>
<td align="left">9PM</td>
<td align="left">Rep2</td>
<td align="left">WT_Rep2</td>
<td align="right">0.8002306</td>
</tr>
<tr class="odd">
<td align="left">ERR793478</td>
<td align="left">KO</td>
<td align="left">9PM</td>
<td align="left">Rep2</td>
<td align="left">KO_Rep2</td>
<td align="right">0.9357273</td>
</tr>
<tr class="even">
<td align="left">ERR793479</td>
<td align="left">WT</td>
<td align="left">5PM</td>
<td align="left">Rep2</td>
<td align="left">WT_Rep2</td>
<td align="right">0.9701294</td>
</tr>
<tr class="odd">
<td align="left">ERR793480</td>
<td align="left">KO</td>
<td align="left">5PM</td>
<td align="left">Rep2</td>
<td align="left">KO_Rep2</td>
<td align="right">0.6653467</td>
</tr>
<tr class="even">
<td align="left">ERR793481</td>
<td align="left">WT</td>
<td align="left">5AM</td>
<td align="left">Rep2</td>
<td align="left">WT_Rep2</td>
<td align="right">1.1412059</td>
</tr>
<tr class="odd">
<td align="left">ERR793482</td>
<td align="left">WT</td>
<td align="left">5AM</td>
<td align="left">Rep3</td>
<td align="left">WT_Rep3</td>
<td align="right">0.8747762</td>
</tr>
<tr class="even">
<td align="left">ERR793483</td>
<td align="left">KO</td>
<td align="left">9AM</td>
<td align="left">Rep3</td>
<td align="left">KO_Rep3</td>
<td align="right">1.1050007</td>
</tr>
<tr class="odd">
<td align="left">ERR793484</td>
<td align="left">WT</td>
<td align="left">5PM</td>
<td align="left">Rep3</td>
<td align="left">WT_Rep3</td>
<td align="right">1.0138664</td>
</tr>
<tr class="even">
<td align="left">ERR793485</td>
<td align="left">KO</td>
<td align="left">9AM</td>
<td align="left">Rep2</td>
<td align="left">KO_Rep2</td>
<td align="right">1.0522057</td>
</tr>
<tr class="odd">
<td align="left">ERR793486</td>
<td align="left">WT</td>
<td align="left">9PM</td>
<td align="left">Rep3</td>
<td align="left">WT_Rep3</td>
<td align="right">0.7389949</td>
</tr>
<tr class="even">
<td align="left">ERR793487</td>
<td align="left">KO</td>
<td align="left">9PM</td>
<td align="left">Rep3</td>
<td align="left">KO_Rep3</td>
<td align="right">1.0855583</td>
</tr>
</tbody>
</table>
<pre class="r"><code>resb &lt;- results(ddsb,name=&quot;time9PM.condKO&quot;)
# same alpha as their paper
tblb &lt;- timecoursePlots(resb,rldb,k=4,alpha=0.05)</code></pre>
<pre><code>## Warning in lda.default(x, grouping, ...): variables are collinear</code></pre>
<p><img src="RNA-seq_files/figure-html/results_BMAL1KO-1.png" /><!-- --><img src="RNA-seq_files/figure-html/results_BMAL1KO-2.png" /><!-- --><img src="RNA-seq_files/figure-html/results_BMAL1KO-3.png" /><!-- --></p>
<pre class="r"><code>#showDT(tblb %&gt;% filter(abs(log2FoldChange) &gt; 1)) </code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
